OpenTabletDriver.Desktop/Reflection/DesktopPluginManager.cs

```diff
@@ -135,54 +119,91 @@ where IsPluginType(type)
+
+            var name = file.Name.Replace(file.Extension, string.Empty);
+            var tempDir = new DirectoryInfo(Path.Join(TemporaryDirectory.FullName, name));
+            if (!tempDir.Exists)
+                tempDir.Create();
+
+            var pluginPath = Path.Join(AppInfo.Current.PluginDirectory, name);
+            var pluginDir = new DirectoryInfo(pluginPath);
+            switch (file.Extension)
             {
                 case ".zip":
                 {
-                    var path = Path.Join(AppInfo.Current.PluginDirectory, fileInfo.Name.Replace(".zip", string.Empty));
-                    Log.Write("Plugin", $"Installing plugin zip: '{fileInfo.Name}'");
-                    if (Directory.Exists(path))
-                    {
-                        return PluginStateManager.QueueUpdate(filePath);
-                    }
-                    else
-                    {
-                        ZipFile.ExtractToDirectory(filePath, path, true);
-                        return PluginStateResult.Installed;
-                    }
+                    Log.Write("Plugin", $"Installing plugin zip: '{file.Name}'");
+                    ZipFile.ExtractToDirectory(file.FullName, tempDir.FullName, true);
+                    break;
                 }
                 case ".dll":
                 {
-                    Log.Write("Plugin", $"Installing plugin dll: '{fileInfo.Name}'");
-                    if (File.Exists(Path.Join(AppInfo.Current.PluginDirectory, filePath)))
-                    {
-                        return PluginStateManager.QueueUpdate(filePath);
-                    }
-                    else
-                    {
-                        var name = Path.GetFileName(filePath);
-                        var dest = Path.Join(AppInfo.Current.PluginDirectory, name);
-                        File.Copy(filePath, dest, true);
-                        return PluginStateResult.Installed;
-                    }
+                    Log.Write("Plugin", $"Installing plugin dll: '{file.Name}'");
+                    file.CopyTo(Path.Join(tempDir.FullName, file.Name));
+                    break;
                 }
                 default:
-                    return PluginStateResult.Error;
+                    throw new InvalidOperationException($"Unsupported archive type: {file.Extension}");
             }
+            var context = Plugins.FirstOrDefault(ctx => ctx.Directory == pluginDir);
+            var result = pluginDir.Exists ? UpdatePlugin(context, tempDir) : InstallPlugin(pluginDir, tempDir);
+
+            if (!TemporaryDirectory.GetFileSystemInfos().Any())
+                Directory.Delete(TemporaryDirectory.FullName, true);
+            return result;
+        }
+
+        public bool InstallPlugin(DirectoryInfo target, DirectoryInfo source)
+        {
+            source.MoveTo(target.FullName);
+            return true;
+        }
+
+        public bool UninstallPlugin(DesktopPluginContext plugin)
```

Comment:
During testing UX crashed on this function with the RPC exception:

```
System.NullReferenceException: Object reference not set to an instance of an object.
      at OpenTabletDriver.Desktop.Reflection.DesktopPluginManager.UninstallPlugin(DesktopPluginContext plugin)
      at OpenTabletDriver.Desktop.Reflection.DesktopPluginManager.UpdatePlugin(DesktopPluginContext plugin, DirectoryInfo source)
      at OpenTabletDriver.Desktop.Reflection.DesktopPluginManager.InstallPlugin(String filePath)
      at OpenTabletDriver.Daemon.DriverDaemon.InstallPlugin(String filePath)
```

Which happened during an update procedure of a previously uninstalled plugin.

Steps to reproduce:

1. Install Plugin.
2. Uninstall Plugin.
3. Install Plugin, and then install it again to trigger update.

Reply:
Fixed